ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -253,6 +264,19 @@ long getLastModifiedTime() {
   }
 
   /**
+   * The resetKey for the index segment.
+   * @return the reset key for the index segment which is a {@link Pair} of StoreKey and
+   * {@link PersistentIndex.IndexEntryType}
+   */
+  Pair<StoreKey, PersistentIndex.IndexEntryType> getResetKey() {
+    if (version == (short) 1 && resetKey != null) {
```

Comment:
right now, the tests uses the existing IndexSegment even for testing version_0. Just that for persisting, a different method is used and not the writeIndexSegmentToFile() within actual IndexSegment as it support writing in version_0. Else, I have to come up with another class altogether for version 0 which pretty much a copy of existing IndexSegment except for the resetKey setting, lastModifiedTime and persistence.